### PR TITLE
Enhance the details pane

### PR DESCRIPTION
**Pull request by peterbourgon**
This PR will enhance the details struct (and thereby the details pane) with details from the origin atom (process, IP), specifically stuff like cgroup names, Docker container info, etc.

### CODE REVIEW

**Comment by peterbourgon** [comment]
Addresses #21

**Comment by peterbourgon** [comment]
screen shot 2015-05-21 at 19 03 44

**Comment by peterbourgon** [comment]
Steps to get a piece of data into the detail pane:

1. In the probe, add the data to the NodeMetadata, either [directly](https://sourcegraph.com/github.com/weaveworks/scope@c193d69b19918fcbb4540449627c0e3747b40dfc/.tree/probe/spy.go#startline=82&endline=82) or [with a ProcessMapper](https://sourcegraph.com/github.com/weaveworks/scope@c193d69b19918fcbb4540449627c0e3747b40dfc/.tree/probe/docker_process_mapper.go#startline=100&endline=111). Note which topology it's being added to; probably it will be the Process (Applications) topology.
2. In the app, [map the data to a table row](https://sourcegraph.com/github.com/weaveworks/scope@a1e8143f9b07b795c025ac233b55eebacaf3bde5/.tree/app/origin_node.go#startline=36&endline=38).

**Comment by peterbourgon** [comment]
@tomwilkie ready for review, I think. This isn't a beautiful implementation but it'll do.

File: app/origin_node.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	return originNodeForNetworkHost(node), true
+}
+
+func originNodeForProcess(node report.NodeMetadata) OriginNode {
+	rows := []report.Row{
+		{Key: "Host", ValueMajor: node["domain"], ValueMinor: ""},
+		{Key: "PID", ValueMajor: node["pid"], ValueMinor: ""},
+		{Key: "Process name", ValueMajor: node["name"], ValueMinor: ""},
+	}
+	if val, ok := node["docker_id"]; ok {
```
Comment (tomwilkie) [inline app/origin_node.go]: Would this be better as a table?  ie

```
for key, human := range map[string]string{
    "docker_id":  "Container ID",
    "docker_name": "Container Name",
    "cgroup": "cgroup",
} {
    if val, ok := node[key]; ok {
        rows = append(rows, report.Row{Key: human, ValueMajor: val, ValueMinor: ""})
    }
}
```

File: app/static.go [hunk 2]
```diff
@@ -11,15 +11,15 @@ import (
 	"time"
```
Comment (tomwilkie) [inline app/static.go]: I'd leave this file out of this change.  Just push this separately without a review.

File: circle.yml [hunk 3]
```diff
@@ -26,6 +26,8 @@ dependencies:
   post:
     - go clean -i net
     - go install -tags netgo std
+    - go get github.com/davecgh/go-spew/spew
+    - go get github.com/pmezard/go-difflib/difflib
```
Comment (tomwilkie) [inline circle.yml]: If PR #85 gets in first, these have moved to the Makefile.

**Comment by tomwilkie** [comment]
Other than that one comment, LGTM.  Will definitely need a rebase and squash - we don't want merge commits, gofmt fixes, circle debugging etc.  I would squash into one.

File: app/origin_node.go (hunk 1 above)
Comment (peterbourgon) [inline app/origin_node.go]: Can do, but is it really better? I don't know. I kind of like it as-is, declaratively. But I'm a 2/10 on it, so if you feel more strongly, just say so and I change it.

Comment (tomwilkie) [inline app/origin_node.go]: I think I'd prefer to push the human names into the node map and ultimately have the come from the process mappers.  But as is its slightly easier to follow (and extend) as a loop like this, I think.  I'm thinking we're going to be adding a lot more in here (docker image id and name, for instance).

File: circle.yml (hunk 3 above)
Comment (tomwilkie) [inline circle.yml]: Actually do these need to be go gotten manually?  Wouldn't it be better to update the go get cmd in the makefile to fetch test dependancies, or was that not possible?

**Comment by tomwilkie** [comment]
Problem with the table approach I suggested; ordering is not stable.

**Comment by peterbourgon** [comment]
D'oh. Will fix.

**Pull request merged**